src/routers/user.ts

```diff
@@ -1,24 +1,51 @@
 import express from 'express';
+import { create, read, update, del } from '../operators/UserOperations';
 
 const userRouter = express.Router();
+const userError = 'User not found.';
 
-userRouter.get('/', (req, res) => {
-	console.log('Get User');
-	res.send('Get userRouter');
+userRouter.get('/:id', async (req, res) => {
+	try {
+		console.log('Get User');
+		const user = await read(req.params.id);
+		res.send({ ...user, semesters: Array.from(user?.semesters || []) });
+	} catch (e) {
+		res.status(404).send(userError);
+		console.error(`Error: ${e} - Status Code ${res.statusCode}`);
+	}
 });
 
-userRouter.post('/', (req, res) => {
-	console.log('Post User');
-	res.send('Post userRouter');
+userRouter.post('/', async (req, res) => {
+	try {
+		console.log('Post User');
+		await update(req.body);
+		res.send('User posted');
+	} catch (e) {
+		res.status(404).send(userError);
+		console.error(`Error: ${e} - Status Code ${res.statusCode}`);
+	}
 });
 
-userRouter.put('/', (req, res) => {
-	console.log('Put User');
-	res.send('Put userRouter');
+userRouter.put('/', async (req, res) => {
+	try {
+		console.log('Put User');
+		await create(req.body);
+		res.send('User Created');
+	} catch (e) {
+		res.status(404).send(userError);
+		console.error(`Error: ${e} - Status Code ${res.statusCode}`);
+	}
 });
 
-userRouter.delete('/', (req, res) => {
-	console.log('Delete User');
-	res.send('Delete userRouter');
+userRouter.delete('/', async (req, res) => {
+	try {
+		console.log('Delete User');
+		await del(req.body.id);
```

Comment:
It won't be the req.body.id, it will be the req.userId I think. You may need to rebase to get this.